plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java

```diff
@@ -38,15 +41,51 @@
 import org.eclipse.jface.dialogs.IDialogConstants;
 import org.eclipse.jface.dialogs.MessageDialog;
 import org.eclipse.swt.widgets.Shell;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
 
 public class StandardDeployCommandHandler extends DeployCommandHandler {
+  private static final Logger logger = Logger.getLogger(StandardDeployCommandHandler.class.getName());
 
   public StandardDeployCommandHandler() {
     super(AnalyticsEvents.APP_ENGINE_DEPLOY_STANDARD);
   }
 
   @Override
   protected boolean checkProject(Shell shell, IProject project) throws CoreException {
+    return checkJspConfiguration(shell, project)
+        && checkAppEngineRuntimeCompatibility(shell, project);
+  }
+
+  private boolean checkAppEngineRuntimeCompatibility(Shell shell, IProject project) {
+    IFacetedProject facetedProject = null;
```

Comment:
To avoid splitting declaration and initialization, include more in the try block

Reply:
`ProjectFacetsManager.create(project)` can return `null`, so it's a condition we need to handle.

Reply:
Yes, you can handle it inside the try block.